lib/log_int.h

```diff
@@ -52,6 +52,8 @@ extern const char *zlog_priority[];
 /* Generic function for zlog. */
 extern void vzlog(int priority, const char *format, va_list args);
 extern void zlog(int priority, const char *format, ...) PRINTF_ATTRIBUTE(2, 3);
+#define flog(priority, ferr_id, format, ...)                                   \
+	zlog(priority, "[EC %" PRIu32 "] " format, ferr_id, ##__VA_ARGS__)
```

Comment:
interesting - I hadn't noticed the split between `log.h` and `log_int.h` before this. it seems as if the purpose of `log_int.h` was to be 'internal' at some point, maybe? I sort of wish both `zlog()` and your new `flog()` were in `log.h` - it would be clearer to me if the general, useful apis were all in `log.h`

Reply:
Yeah I agree with you. I think `zlog()` and `flog()` shouldn't be hidden as they can be useful in a few selected cases (when the log priority is a variable). I'll update this PR later unless someone objects.

I believe `zlog()` was moved to _lib/log_int.h_ because some people were writing code like `zlog(LOG_ERR, ...)` when they could use `zlog_err()` directly. So the idea was probably to discourage this kind of style.

Reply:
@mjstapp just pushed an update moving zlog() back to the public logging API.